node/src/components/storage.rs

```diff
@@ -177,13 +142,12 @@ impl<B: Value> Default for DeployMetadata<B> {
 /// If this trait is ultimately only used for testing scenarios, we shouldn't need to expose it to
 /// the reactor - it can simply use a concrete type which implements this trait.
 pub trait StorageType {
-    type Block: Value;
+    type Block: Value + WithBlockHeight;
```

Comment:
I think this is a bit like cheating. 

Originally, constraints on `type Block` didn't have any domain knowledge. `Value` is generalized enough that it can be used for anything that has `Id` and `Header`. Now, `WithBlockHeight` is introducing domain knowledge to this trait. It's conflating two worlds and, in my opinion, is decreasing the power to abstract over things.

Reply:
I think the abstractions we have in the storage component are already needlessly painful.  I'd much rather we just got rid of the `Value` trait, and the `Store` trait and made the component specific to our needs.

In the meantime, I don't think using extension traits to extend a `Value` to fit a block's requirements is a bad thing.  Blocks have a height, deploys don't - if we insist on keeping things generic and not just using actual `Block` and `Deploy` types then I think this extension trait to provide a getter for a block's height is fine personally.